fuel-vm/src/interpreter/memory/allocation_tests.rs

```diff
@@ -109,18 +109,19 @@ fn test_memeq(b: Word, c: Word, d: Word) -> Result<(), RuntimeError> {
     Ok(())
 }
 
-#[test_case(true, 20, 40, 10 => Ok(()); "Can move sp up")]
-#[test_case(false, 20, 40, 10 => Ok(()); "Can move sp down")]
-#[test_case(true, u64::MAX - 10, u64::MAX, 20 => Err(RuntimeError::Recoverable(PanicReason::MemoryOverflow)); "Panics on overflowing addition")]
-#[test_case(false, 10, 11, 20 => Err(RuntimeError::Recoverable(PanicReason::MemoryOverflow)); "Panics on underflowing subtraction")]
-#[test_case(true, u64::MAX, u64::MAX, 0 => Err(RuntimeError::Recoverable(PanicReason::MemoryOverflow)); "Panics on equality check for overflowing addition")]
-#[test_case(false, 0, u64::MAX, 1 => Err(RuntimeError::Recoverable(PanicReason::MemoryOverflow)); "Panics on zero check for underflowing subtraction")]
-fn test_stack_pointer_overflow(add: bool, mut sp: Word, hp: Word, v: Word) -> Result<(), RuntimeError> {
+#[test_case(true, 20, 0, 40, 10 => Ok(()); "Can move sp up")]
+#[test_case(false, 20, 0, 40, 10 => Ok(()); "Can move sp down")]
+#[test_case(true, u64::MAX - 10, 0, u64::MAX, 20 => Err(RuntimeError::Recoverable(PanicReason::MemoryOverflow)); "Panics on overflowing addition")]
+#[test_case(false, 10, 0, 11, 20 => Err(RuntimeError::Recoverable(PanicReason::MemoryOverflow)); "Panics on underflowing subtraction")]
+#[test_case(true, u64::MAX, 0, u64::MAX, 0 => Err(RuntimeError::Recoverable(PanicReason::MemoryOverflow)); "Panics on equality check for overflowing addition")]
+#[test_case(false, 0, 0, u64::MAX, 1 => Err(RuntimeError::Recoverable(PanicReason::MemoryOverflow)); "Panics on zero check for underflowing subtraction")]
+fn test_stack_pointer_overflow(add: bool, mut sp: Word, ssp: Word, hp: Word, v: Word) -> Result<(), RuntimeError> {
```

Comment:
It would be nice to add a new test case=)

Reply:
Done